packages/legend-query-builder/src/components/QueryBuilderPropertyExpressionEditor.tsx

```diff
@@ -108,6 +119,78 @@ const DerivedPropertyParameterValueEditor = observer(
       }),
       [handleDrop],
     );
+    const queryBuilderState = derivedPropertyExpressionState.queryBuilderState;
+    // Resets the next property expression in the property chain for milestoned properties when the user changes
+    // the milestoned dates of current property expression and it propagates those dates to next property expression.
+    const resetNextExpression = (
+      nextExpression: AbstractPropertyExpression,
+    ): void => {
+      const milestoningStereotype = getMilestoneTemporalStereotype(
+        guaranteeType(
+          nextExpression.func.value.genericType.value.rawType,
+          Class,
+        ),
+        queryBuilderState.graphManagerState.graph,
+      );
+      nextExpression.parametersValues.slice(1).forEach((parameter, index) => {
+        if (
+          milestoningStereotype &&
+          parameter instanceof INTERNAL__PropagatedValue &&
+          !matchMilestoningParameterValue(
+            milestoningStereotype,
+            index,
+            parameter.getValue(),
+            queryBuilderState.milestoningState,
+          )
+        ) {
+          const newParameterValue = new INTERNAL__PropagatedValue(() =>
+            guaranteeNonNullable(
+              guaranteeNonNullable(
+                queryBuilderState.milestoningState.milestoningImplementations.find(
+                  (impl) => impl.stereotype === milestoningStereotype,
+                ),
+              ).getMilestoningDate(index),
+            ),
+          );
+          newParameterValue.isPropagatedValue = false;
+          functionExpression_setParameterValue(
+            guaranteeType(nextExpression, AbstractPropertyExpression),
+            guaranteeNonNullable(newParameterValue),
+            index + 1,
+            queryBuilderState.observableContext,
+          );
+        }
+      });
+    };
+    const checkDatePropagation = (
+      nextExpression: ValueSpecification | undefined,
+    ): void => {
+      if (
+        nextExpression instanceof AbstractPropertyExpression &&
+        isDefaultDatePropagationSupported(nextExpression, queryBuilderState) &&
+        nextExpression.func.value.genericType.value.rawType instanceof Class
+      ) {
+        queryBuilderState.applicationStore.setActionAlertInfo({
+          message:
+            'You have just changed a milestoning date in the property expression chain, this date will be propagated down the rest of the chain. Do you want to proceed? Otherwise, you can choose to propagate the global milestoning dates instead.',
```

Comment:
I don't like the mention of "global". Seems more of a code term.
Is it right to say "current" or the Main class date or is that not accurate

Reply:
May be default milestoning dates?